### PR TITLE
Added changes to facilitate health information exchange between HIU and HIP.

### PR DESCRIPTION
**Pull request by atulai-sg**

### CODE REVIEW

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/HealthInformationService.java [hunk 1]
```diff
@@ -124,12 +124,44 @@ public void healthInformation(HIPHealthInformationRequest hipHealthInformationRe
     log.debug(
         "health information acknowledgment request body : "
             + onHealthInformationRequest.toString());
+    // Acknowledge to gateway that health information request has been received.
     healthInformationAcknowledgementRequest(
         hipHealthInformationRequest, onHealthInformationRequest);
+    // Prepare health information bundle request which needs to be sent to HIP.
```
Comment (chandrashekar-s) [inline src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/HealthInformationService.java:130]:
```suggestion
    // Prepare health information bundle request which needs to be sent to HIU.
```

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/HealthInformationService.java [hunk 2]
```diff
@@ -124,12 +124,44 @@ public void healthInformation(HIPHealthInformationRequest hipHealthInformationRe
...
     HealthInformationBundle healthInformationBundle =
-        fetchHealthInformationBundle(hipHealthInformationRequest);
-    pushHealthInformation(healthInformationBundle, consentId);
-    healthInformationPushAcknowledge(hipHealthInformationRequest, consentId);
+        fetchHealthInformationBundle(hipHealthInformationRequest, gatewayRequestId);
```
Comment (chandrashekar-s) [inline src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/HealthInformationService.java:132]: We might have to validate if the consent has status in GRANTED state, and also check for the date range etc.

**Review by chandrashekar-s — COMMENTED** [review]
Thanks @atulai-sg and @VenuAjitesh for working on this changes. This is a great progress in less time. I had a broader look at the PR and it looks fine. There are only few minor comments, please have a look. In general, ee could provide a bit more comments i.e. javadoc could be improved.

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/HealthInformationService.java [hunk 3]
```diff
@@ -124,12 +124,44 @@ public void healthInformation(HIPHealthInformationRequest hipHealthInformationRe
...
+    // Request for health information bundle from HIP.
```
[the remark this answers is not in the document]
Comment (atulai-sg) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/HealthInformationService.java:133]: ack

File: src/main/java/com/nha/abdm/wrapper/hip/hrp/database/mongo/tables/ConsentCareContextMapping.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.nha.abdm.wrapper.hiu.hrp.consent.requests.ConsentCareContexts;
+import java.util.List;
+import lombok.Data;
+import org.springframework.data.mongodb.core.index.Indexed;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+@Data
+@Document(collection = "consent-careContexts")
```
[the remark this answers is not in the document]
Comment (atulai-sg) [reply src/main/java/com/nha/abdm/wrapper/hip/hrp/database/mongo/tables/ConsentCareContextMapping.java:12]: I have added an issue to tackle this: https://github.com/NHA-ABDM/ABDM-wrapper/issues/78

**Review by chandrashekar-s — APPROVED** [review]
Thanks for the changes @atulai-sg

**Pull request merged**